### PR TITLE
Add LIBSPDM_ENABLE_CAPABILITY_MUT_AUTH_CAP check for FINISH

### PR DESCRIPTION
**Pull request by jyao1**
That can save ~5K space for the device, to remove X.509 process.

Signed-off-by: Jiewen Yao <jiewen.yao@intel.com>

Size before: (X64, VS2019, release, mbedtls, 
cap: disable CHAL_CAP/PSK_EX_CAP/GET_CSR_CAP/SET_CERTIFICATE_CAP/CHUNK_CAP/MUT_AUTH_CAP/ENCAP_CAP,
algo: only enable SHA384/ECDSA/ECDH/AES-GCM

```
10/30/2022  10:44 AM            94,208 test_size_of_spdm_requester.exe
10/30/2022  10:44 AM            65,536 test_size_of_spdm_responder.exe
```

After fix mut_auth

```
10/30/2022  11:23 AM            94,208 test_size_of_spdm_requester.exe
10/30/2022  11:23 AM            59,392 test_size_of_spdm_responder.exe
```

### CODE REVIEW

File: library/spdm_requester_lib/libspdm_req_finish.c [hunk 1]
```diff
@@ -370,15 +372,16 @@ static libspdm_return_t libspdm_try_send_receive_finish(libspdm_context_t *spdm_
 
     spdm_request->header.spdm_version = libspdm_get_connection_version (spdm_context);
     spdm_request->header.request_response_code = SPDM_FINISH;
+    spdm_request->header.param1 = 0;
+    spdm_request->header.param2 = 0;
+    signature_size = 0;
     if (session_info->mut_auth_requested) {
```
Comment (steven-bellock) [inline library/spdm_requester_lib/libspdm_req_finish.c:378]: There's nothing in `libspdm_req_key_exchange.c` that checks the Requester's capabilities, either `LIBSPDM_ENABLE_CAPABILITY_MUT_AUTH_CAP` or in the SPDM context, for mutual authentication. It just blindly sets `session_info->mut_auth_requested` to whatever the Responder tells it, even if it doesn't support mutual authentication. That should be fixed in this pull request.

File: library/spdm_requester_lib/libspdm_req_finish.c [hunk 2]
```diff
@@ -370,15 +372,16 @@ static libspdm_return_t libspdm_try_send_receive_finish(libspdm_context_t *spdm_
...
+#if LIBSPDM_ENABLE_CAPABILITY_MUT_AUTH_CAP
```
Comment (steven-bellock) [inline library/spdm_requester_lib/libspdm_req_finish.c:379]: This line can go before `if (session_info->mut_auth_requested) {`

**Review by steven-bellock — CHANGES_REQUESTED** [review]

File: library/spdm_requester_lib/libspdm_req_finish.c (hunk 2 above)
Comment (jyao1) [reply library/spdm_requester_lib/libspdm_req_finish.c:378]: done.

File: library/spdm_requester_lib/libspdm_req_finish.c (hunk 1 above)
Comment (jyao1) [reply library/spdm_requester_lib/libspdm_req_finish.c]: add check in req_key_exchange.

**Review by steven-bellock — APPROVED** [review]

**Pull request merged**